crates/btc-relay/src/lib.rs

```diff
@@ -903,6 +962,11 @@ impl<T: Config> Pallet<T> {
 
         Self::store_rich_header(basic_block_header.clone(), block_height, blockchain.chain_id);
 
+        // Add chainwork for block if prev block contains chainwork
+        if Self::contains_chainwork(basic_block_header.hash_prev_block) {
+            Self::_set_chainwork_for_block(basic_block_header)?;
```

Comment:
the `if` condition is required as after migration there will be bitcoin blocks that will not have a chainwork, the relayer would require some block time to successfully sync the chainwork for the latest BTC block.